### PR TITLE
Project init

### PR DESCRIPTION
**Pull request by ShellyXueHan**
1. setup api and web
2. dockerize app

/bot-ignore-length

### CODE REVIEW

File: api/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      "ecmaVersion": 2018,
+      "sourceType": "module"
+  },
+  "rules": {
+    "prettier/prettier": ["error"],
+    "strict": 0,
+    "linebreak-style": 0,
+    "indent": ["error", 2]
+  }
+}
```
Comment (patricksimonian) [inline api/.eslintrc]: new line

File: api/dev.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM node:11.0.0
```
Comment (patricksimonian) [inline api/dev.Dockerfile]: why is dev docker file a higher version of Node?

File: web/dev.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM node:11.0.0
```
Comment (patricksimonian) [inline web/dev.Dockerfile]: again why is node version higher?

File: web/public/index.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico" />
```
Comment (patricksimonian) [inline web/public/index.html]: it is a good practice to have a `<meta name="description" content="foo" />` at a minimum in the head. I'd recommend using a library like `react-helmet` to achieve this

File: web/src/App.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
```
Comment (patricksimonian) [inline web/src/App.js]: i'd recommend to changing this to a css module instead of using global css

File: web/src/App.test.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
```
Comment (patricksimonian) [inline web/src/App.test.js]: i'd reccomend using react-testing-library as it handles the latest version of react 16.8

File: web/src/index.css [hunk 7]
```diff
@@ -0,0 +1,14 @@
+body {
+  margin: 0;
+  padding: 0;
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
```
Comment (patricksimonian) [inline web/src/index.css]: i'd recommend using Typography.js for your global fonts.

File: web/src/App.js (hunk 5 above)
Comment (ShellyXueHan) [reply web/src/App.js]: oh yea iI haven't changed anything yet within the app. Going to setup the pipeline first before going there.

File: web/src/App.test.js (hunk 6 above)
Comment (ShellyXueHan) [reply web/src/App.test.js]: sure! will do in later work as well.

File: web/src/App.js (hunk 5 above)
Comment (ShellyXueHan) [reply web/src/App.js]: updated

File: web/src/App.test.js (hunk 6 above)
Comment (ShellyXueHan) [reply web/src/App.test.js]: updated

File: web/src/components/App.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+class App extends Component {
+  render() {
+    return (
+      <div className={styles.app}>
+        <Helmet>
+          <meta charSet="utf-8" />
+          <title>Realm-o-Matic</title>
+          <meta name="description" content="Self-service KeyCloak Realm Provosioner" />
+          <meta name="twitter:author" content="ShellyXueHan" />
+          <meta name="twitter:description" content="Self-service KeyCloak Realm Provosioner"
```
Comment (patricksimonian) [inline web/src/components/App.js]: yyou should have a logo for the realm-o-matic :) `twitter:image`

Comment (ShellyXueHan) [reply web/src/components/App.js]: will do when I have one :P

**Pull request merged**